src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -154,10 +154,41 @@ export class QueryStringManager {
    */
   public setQuery = (query: Partial<Query>) => {
     const curQuery = this.query$.getValue();
-    const newQuery = { ...curQuery, ...query };
+    let newQuery = { ...curQuery, ...query };
     if (!isEqual(curQuery, newQuery)) {
-      // Add to recent datasets if dataset has changed
+      // Check if dataset changed and if new dataset has language restrictions
       if (newQuery.dataset && !isEqual(curQuery.dataset, newQuery.dataset)) {
+        // Get supported languages for the dataset
+        const supportedLanguages = this.datasetService
+          .getType(newQuery.dataset.type)
+          ?.supportedLanguages(newQuery.dataset);
+
+        // If we have supported languages and current language isn't supported
+        if (supportedLanguages && !supportedLanguages.includes(newQuery.language)) {
+          // Get initial query with first supported language and new dataset
+          newQuery = this.getInitialQuery({
+            language: supportedLanguages[0],
+            dataset: newQuery.dataset,
+          });
+
+          // Show warning about language change
+          showWarning(this.notifications, {
```

Comment:
Shouldn't this just be an info?

Reply:
+1

Reply:
it could be show warning is the helper function for this manager that passes an id that prevents duplicate notifications from popping up. so if the user constantly switches between recent data and somehow the current language is always invalid it won't keep adding a notification on the right side and keep pushing the duplicate ones upwards. it will just have a single notification available.

Reply:
<img width="379" alt="Screenshot 2024-10-30 at 12 59 26 PM" src="https://github.com/user-attachments/assets/e37875ea-5c0c-4188-b5ec-a3f7aa481c6a">